src/app/contribution-graph/contribution-graph.component.ts

```diff
@@ -45,4 +48,25 @@ export class ContributionGraphComponent implements OnInit {
     return Array.from(Array(row).keys());
   }
 
+  calcDays(startDateString: string) {
+    const dayTable = [
+      'Sun',
+      'Mon',
+      'Tue',
+      'Wed',
+      'Thu',
+      'Fri',
+      'Sat'
+    ];
+    const days = [];
+    const startDate = new Date(startDateString);
+    const startDay = startDate.getDay(); // Sunday - Saturday : 0 - 6
+    for (let i = 0; i < 4; i++) {
+      const mIndex = (startDay + i * 2) % 7;
+      days[i] = dayTable[mIndex];
```

Comment:
you could merge this with the line above doing:
`days[i] = (startDay + i * 2) % 7;`

This will make the code more concise and easier to read.